src/runtime_src/core/edge/drm/zocl/cu_scu.c

```diff
@@ -76,13 +77,12 @@ static int scu_peek_credit(void *core)
 static void scu_xgq_start(struct xrt_cu_scu *scu, u32 *data)
 {
 	struct xgq_cmd_start_cuidx *cmd = (struct xgq_cmd_start_cuidx *)data;
-	u32 num_reg = 0;
 	u32 i = 0;
 	u32 *cu_regfile = scu->vaddr;
 
-	num_reg = (cmd->hdr.count - (sizeof(struct xgq_cmd_start_cuidx)
+	scu->num_reg = (cmd->hdr.count - (sizeof(struct xgq_cmd_start_cuidx)
```

Comment:
How are you guaranteeing that cmd->hdr.count will always be larger then the size of the structure.   If this should occur, then band things are going to happen.